### PR TITLE
Add support for PR regex filters per team

### LINKED ISSUE #2: Add support for PR regex filters per team
Examples:

- [aks] Update nginx-ingress-controller to version 0.32.0 in somewhere
- Dependabot: Update 19 dependencies in /something/bla

### PR DESCRIPTION
**Pull request by paxos**
Closes #2

### CODE REVIEW

File: lib/devops.ts [hunk 1]
```diff
@@ -43,6 +43,33 @@ export abstract class DevOpsBaseClient implements IDevOpsClient {
 
   abstract async fetchPullRequests(): Promise<any>;
 
+  filterPRs(devOpsResponse: DevOpsResponse): DevOpsResponse {
+    if (!this.devOpsTeam.filters) {
```
Comment (matthiaswenz) [inline lib/devops.ts:47]: Probably could also be 
```suggestion
    if (!this.devOpsTeam.filters && this.devOpsTeam.filters.length > 0) {
```
to check for zero length filters

File: lib/devops.ts [hunk 2]
```diff
@@ -43,6 +43,33 @@ export abstract class DevOpsBaseClient implements IDevOpsClient {
...
+      return devOpsResponse;
+    }
+
+    const result = { ...devOpsResponse };
+    result.value = [];
+
+    for (let PR of devOpsResponse.value) {
```
Comment (matthiaswenz) [inline lib/devops.ts:54]: Style suggestion: `pr` here for variable.

File: lib/devops.ts [hunk 3]
```diff
@@ -43,6 +43,33 @@ export abstract class DevOpsBaseClient implements IDevOpsClient {
...
+      let filterHit = false;
+      for (let filter of this.devOpsTeam.filters) {
+        let regex = RegExp(filter);
```
Comment (matthiaswenz) [inline lib/devops.ts:57]: In `IDevOpsTeam.filters` is already specified to be `RegExp[]` so do we need to wrap this in another `RegExp(...)` here?

File: config/config.ts [hunk 4]
```diff
@@ -24,6 +24,7 @@ export interface IDevOpsTeam {
   collection: string;
   project: string;
   repository: string;
+  filters?: RegExp[]; // Matches pull requests whos names match one of those regex
```
Comment (matthiaswenz) [inline config/config.ts:27]: > Matches pull requests
assumes that the filter is exclusive according to the code - so maybe changing this to 
```suggestion
  filters?: RegExp[]; // Excludes pull requests who's title matches one of the regular expressions
```

Comment (paxos) [reply config/config.ts]: 🙏

File: lib/devops.ts (hunk 1 above)
Comment (octopotato) [reply lib/devops.ts:47]: do you mean `&& this.devOpsTeam.filters.length === 0)`?

Comment (matthiaswenz) [reply lib/devops.ts:47]: A yeah, double negative.

Comment (paxos) [reply lib/devops.ts]: Best of all worlds https://github.com/paxos/motivatron/pull/3/commits/fb2c1272c07149bb0d4fc972c1f88a6b9600f977:
```
 if (!Array.isArray(this.devOpsTeam.filters)) {
```

File: lib/devops.ts (hunk 3 above)
Comment (paxos) [reply lib/devops.ts]: Good point: I had it coming back as a string in my initial tests, but I guess my debugger was lying to me. Replaced it with a much better implementation.

Comment (paxos) [reply lib/devops.ts]: In https://github.com/paxos/motivatron/pull/3/commits/772843581da5f00eda2acd2e2e0d0fa5d0f6770b

**Comment by paxos** [comment]
New output:
`1 PR is waiting for review (1 filtered). …`

**Pull request merged**